### ISSUE TITLE
Packaging Dino for ArchLinux

### ISSUE THREAD

**Issue by NicoHood** [dino/dino#65]
Hello,
I really love the dino chat client, especially because it supports OMEMO #36 really well. I'd love to bring it to the [community] repository of ArchLinux. However I have some request for that first:

* Can you please tag a release?
* Can you please [GPG sign](https://github.com/NicoHood/gpgit) your releases?
* You have some dependencies on submodules like libsignal-protocol-c. Is there a way to build Dino with shared libraries? On ArchLinux we avoid static linkage because of security reasons and minor size reasons.

Keep up the good work and thanks for sharing!

**Comment by jubalh** [comment]
> Can you please GPG sign your releases?

Glad someone will help me convince the whisper guys, see https://github.com/WhisperSystems/libsignal-protocol-c/issues/60 ;-)

> You have some dependencies on submodules like libsignal-protocol-c. Is there a way to build Dino with shared libraries? On ArchLinux we avoid static linkage because of security reasons and minor size reasons.

See: https://github.com/dino/dino/issues/22 and https://github.com/WhisperSystems/libsignal-protocol-c/issues/59

**Comment by NicoHood** [comment]
Thanks, opened a new issue at https://github.com/WhisperSystems/libsignal-protocol-c/issues/68
They should really sign their releases at they are one of the most important crypto chat programmers. Cant understand why they are so resistant. Hope to have some better cooperation with the Dino devs :)

**Comment by fiaxh** [comment]
I don't think that it would be a good idea to release Dino just yet. Dino is supposed to be a stable, intuitive and nice-looking client, but its not there yet. Releasing it now would present just yet another XMPP desktop client with a terrible user experience. Please give it some more time ;)

There currently is a [dino-git](https://aur.archlinux.org/packages/dino-git/) package in aur, which I think is a suitable way to distribute Dino in it's early state.

PGP signing the release will be done.

**Comment by NicoHood** [comment]
@fiaxh Thanks, very nice feedback. I totally agree and I love your chat.

I was tricked, because debian has an application with the same name `dino`:
https://aur.archlinux.org/packages/dino/

So for now we will use the git version. I will package dino for ArchLinux once it is ready :) Very nice work guys :)

**Comment by mar-v-in** [comment]
@NicoHood (and @jubalh in case you didn't notice yet): you can build the omemo plugin using a shared libsignal-protocol-c by setting the environment var `SHARED_SIGNAL_PROTOCOL` or when directly invoking cmake via `-DSHARED_SIGNAL_PROTOCOL=yes`.

It might be that we will lift the dependency to libsignal-protocol-c completely before the first stable release: IMHO upstream is not very well managed and the next protocol iteration of OMEMO will no longer use the protobuf output generated by libsignal-protocol-c (and instead directly serialize the messages into XMPP).

**Comment by NicoHood** [comment]
@mar-v-in Thanks for this hint. I think we should keep it as a workaround then, also to avoid another AUR dependency. I'd also love to get rid of their dependencies.

**Comment by jubalh** [comment]
@mar-v-in thanks! Changed the package do use it now.

**Comment by NicoHood** [comment]
@fiaxh Could you please also sign your git commits via GPG? @mar-v-in already does this, which is great :)

**Comment by jubalh** [comment]
@mar-v-in is `SHARED_SIGNAL_PROTOCOL` still used?

**Comment by mar-v-in** [comment]
@jubalh yes, it's used here: https://github.com/dino/dino/blob/master/plugins/signal-protocol/CMakeLists.txt#L31

_Sent from my Eterna Nexus 5X using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
